mssql.go

```diff
@@ -91,6 +92,21 @@ func NewConnector(dsn string) (*Connector, error) {
 	return c, nil
 }
 
+// SecurityTokenProvider implementations are called during federated
+// authentication security token login sequences at the point when the
+// security token is required.  The string returned should be the access
+// token to supply to the server, otherwise an error can be returned to
+// indicate why a token is not available.
+type SecurityTokenProvider func(ctx context.Context) (string, error)
```

Comment:
Ah! I was looking at where you changed the provider func to take in the context.
nicer. thanks.